backend/clubs/serializers.py

```diff
@@ -2424,6 +2425,51 @@ class Meta:
         fields = ("text", "multiple_choice", "question_type", "question")
 
 
+class ApplicationExtensionSerializer(serializers.ModelSerializer):
+    first_name = serializers.CharField(source="user.first_name", read_only=True)
+    last_name = serializers.CharField(source="user.last_name", read_only=True)
+    username = serializers.CharField(source="user.username", read_only=False)
+    graduation_year = serializers.CharField(
+        source="user.profile.graduation_year", read_only=True
+    )
+
+    class Meta:
+        model = ApplicationExtension
+        fields = (
+            "id",
+            "username",
+            "first_name",
+            "last_name",
+            "graduation_year",
+            "end_time",
+        )
+
+    def validate_username(self, value):
+        user = get_user_model().objects.filter(username=value).first()
+        if not user:
+            raise serializers.ValidationError("Please provide a valid username!")
+        return value
+
+    def create(self, validated_data):
+        username = validated_data.get("user").pop("username")
+        validated_data["user"] = get_user_model().objects.get(username=username)
+
+        application_pk = self.context["view"].kwargs.get("application_pk")
+        validated_data["application"] = ClubApplication.objects.filter(
+            pk=application_pk
+        ).first()
+
+        return super().create(validated_data)
+
+    def update(self, instance, validated_data):
+        user_field = validated_data.pop("user", None)
+        if user_field:
+            username = user_field.pop("username")
+            user = get_user_model().objects.get(username=username)
+            instance.user = user
```

Comment:
Why is user not read only? Is there a case where we change the user we give the extension to?

Reply:
If the club admin makes a typo in the user's pennkey, I figured that we should allow them to update the user. However they could also just delete the extension and create a new one, so open to making this read-only. Lmk what you think

Reply:
I was envisioning a UI where the club admin selects users from a dropdown of applicants...so no typo potential. What do you think?

Reply:
Hmm, wouldn't this require the applicant to submit to appear in the system before they can be granted an extension? It's definitely a minor detail but I think it might be better to leave the responsibility solely with the club admin

This is what the UI currently looks like:
<img width="1000" alt="image" src="https://github.com/pennlabs/penn-clubs/assets/60864468/b3255c80-2e34-4ac6-9011-bf2b28d5f81e">

Reply:
I think that makes sense, that's a fair point.

Can we consider emailing applicants that have received extensions letting them know they've been granted one?

Reply:
Sure! I can add that